### PR TITLE
Adding Role Data Source

### PR DESCRIPTION
**Pull request by Aayushi-Bansal**

### CODE REVIEW

File: sumologic/sumologic_role.go [hunk 1]
```diff
@@ -53,6 +53,32 @@ func (s *Client) GetRole(id string) (*Role, error) {
...
+	err = json.Unmarshal(data, &response)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range response.Roles {
+		if r.Name == name {
+			return &r, nil
+		}
+	}
```
Comment (seansain) [inline sumologic/sumologic_role.go:77]: The api has a query param called `name`. We should use that instead of going through all the roles.

Comment (Aayushi-Bansal) [reply sumologic/sumologic_role.go:77]: I searched for it earlier, just like we have for collectors, but wasn't able to find it.
Let me change it, if we really have the support of `name` query parameter in roles API as well.

Comment (seansain) [reply sumologic/sumologic_role.go:77]: Screen Shot 2021-01-15 at 9 27 17 AM
Should be the bottom one here

Comment (Aayushi-Bansal) [reply sumologic/sumologic_role.go]: Incorporated.

**Review by seansain — APPROVED** [review]

**Pull request merged**